### ISSUE TITLE
Error wrong-type-argument stringp

### ISSUE THREAD

**Issue by arichiardi** [syohex/emacs-git-gutter#92]
Hello, I recently updated to:
```
$ emacs --version
GNU Emacs 24.4.1
Copyright (C) 2014 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.
```

And  running tag ```0.83``` I got a problem with the live update:

```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  file-name-nondirectory(nil)
  (let ((file (file-name-nondirectory (git-gutter:base-file))) (now (make-temp-file "git-gutter-cur")) (original (make-temp-file "git-gutter-orig"))) (if (git-gutter:write-original-content original file) (progn (git-gutter:write-current-content now) (git-gutter:start-live-update file original now))))
  (progn (let ((file (file-name-nondirectory (git-gutter:base-file))) (now (make-temp-file "git-gutter-cur")) (original (make-temp-file "git-gutter-orig"))) (if (git-gutter:write-original-content original file) (progn (git-gutter:write-current-content now) (git-gutter:start-live-update file original now)))))
  (if (git-gutter:should-update-p) (progn (let ((file (file-name-nondirectory (git-gutter:base-file))) (now (make-temp-file "git-gutter-cur")) (original (make-temp-file "git-gutter-orig"))) (if (git-gutter:write-original-content original file) (progn (git-gutter:write-current-content now) (git-gutter:start-live-update file original now))))))
  (progn (if (git-gutter:should-update-p) (progn (let ((file (file-name-nondirectory (git-gutter:base-file))) (now (make-temp-file "git-gutter-cur")) (original (make-temp-file "git-gutter-orig"))) (if (git-gutter:write-original-content original file) (progn (git-gutter:write-current-content now) (git-gutter:start-live-update file original now)))))))
  (if (and git-gutter:enabled (buffer-modified-p)) (progn (if (git-gutter:should-update-p) (progn (let ((file (file-name-nondirectory ...)) (now (make-temp-file "git-gutter-cur")) (original (make-temp-file "git-gutter-orig"))) (if (git-gutter:write-original-content original file) (progn (git-gutter:write-current-content now) (git-gutter:start-live-update file original now))))))))
  git-gutter:live-update()
  apply(git-gutter:live-update nil)
  byte-code("r\301\302H\303H\"\210)\301\207" [timer apply 5 6] 4)
  timer-event-handler([t 0 1 0 2 git-gutter:live-update nil idle 0])
```

Has somebody had the same problem? Any hint?

**Comment by syohex** [comment]
Could you show me your configuration about `git-gutter` ?

**Comment by arichiardi** [comment]
Of course, it is basically this (it might be old, I recently updated both ```emacs``` and ```emacs-git-gutter```):

```lisp
(global-git-gutter-mode t)
(git-gutter:linum-setup)

(custom-set-variables
 '(git-gutter:update-interval 2))```

**Comment by syohex** [comment]
I have fixed at #93. Please check latest version.

**Comment by arichiardi** [comment]
Looks ok now. I have not tested if the auto update actually works but I will tell you (visually) soon.
